DSharpPlus/Entities/DiscordChannel.cs

```diff
@@ -573,6 +574,34 @@ public async Task PlaceMemberAsync(DiscordMember member)
                 default, this.Id, null).ConfigureAwait(false);
         }
 
+        /// <summary>
+        /// Follows the news channel
+        /// </summary>
+        /// <param name="targetChannel">Channel to crosspost messages to</param>
+        /// <exception cref="ArgumentException">Thrown when trying to follow a non-news channel</exception>
+        public Task<DiscordFollowedChannel> FollowAsync(DiscordChannel targetChannel)
+        {
+            if (this.Type != ChannelType.News)
+                throw new ArgumentException("Cannot follow a non-news channel.");
+
+            return this.Discord.ApiClient.FollowChannelAsync(this.Id, targetChannel.Id);
+        }
+
+        /// <summary>
+        /// Publish a message in a News channel to following channels
+        /// </summary>
+        /// <param name="message">Message to publish</param>
+        /// <exception cref="UnauthorizedException">
+        ///     Thrown when the current user doesn't have ManageWebhooks and/or SendMessages <see cref="Permissions"/>
```

Comment:
You could probably just link those permission enums directly rather than just saying "ManageWebhooks" and "SendMessages".